plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -250,89 +285,204 @@ func (c *CiscoTelemetryMDT) handleTelemetry(data []byte) {
 		for _, field := range gpbkv.Fields {
 			switch field.Name {
 			case "keys":
-				tags = make(map[string]string, len(field.Fields)+2)
-				tags["source"] = telemetry.GetNodeIdStr()
-				tags["subscription"] = telemetry.GetSubscriptionIdStr()
+				tags = make(map[string]string, len(field.Fields)+3)
+				tags["source"] = msg.GetNodeIdStr()
+				tags["subscription"] = msg.GetSubscriptionIdStr()
+				tags["path"] = msg.GetEncodingPath()
+
 				for _, subfield := range field.Fields {
-					c.parseGPBKVField(subfield, &namebuf, telemetry.EncodingPath, timestamp, tags, nil)
+					c.parseKeyField(tags, subfield, "")
 				}
 			case "content":
-				fields = make(map[string]interface{}, len(field.Fields))
-				for _, subfield := range field.Fields {
-					c.parseGPBKVField(subfield, &namebuf, telemetry.EncodingPath, timestamp, tags, fields)
+				if tags != nil {
```

Comment:
Yes, so far the assumption was there was a "keys"-field followed by a "content"-field which holds true for all datasets I have seen so far. But since it does not hurt I have now made a simple loop which just looks for these 2 fields afterwards a check if both have been found and an early continue if one or both are missing.